setup.py

```diff
@@ -82,6 +82,7 @@ def get_version(package):
     extras_require={
         'dev': ['check-manifest'],
         'test': ['coverage'],
+        'ssl': ['service-identity']
```

Comment:
Why do you require `service-identity`?

Reply:
this is optional dependency not a required one (it will be installed if user specifies extras .e.g, `pip install -e .[ssl]`)

I still keep it in the `setup.py` since Python 3.x will print a warning like this `:0: UserWarning: You do not have a working installation of the service_identity module: 'No module named 'service_identity''.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.` if `service-identity` is not installed.

Reply:
in this case `service-identity` should be mandatory

Reply:
Moved it to required dependencies.